### PR TITLE
Add v4l2_buffer timestamp and sequence

### PR DESCRIPTION
**Pull request by PeterBee97**
## Summary
Added timestamp and sequence number to captured frame.

### CODE REVIEW

File: arch/sim/src/sim/sim_video.c [hunk 1]
```diff
@@ -336,11 +340,14 @@ int sim_video_uninitialize(void)
...
   int ret;
 
   if (priv->next_buf)
     {
       ret = host_video_dqbuf(priv->vdev, priv->next_buf, priv->buf_size);
+      clock_gettime(CLOCK_MONOTONIC, &ts);
+      TIMESPEC_TO_TIMEVAL(priv->next_ts, &ts);
       if (ret > 0)
         {
           priv->capture_cb(0, ret);
```
Comment (xiaoxiang781216) [inline arch/sim/src/sim/sim_video.c:353]: le's extend capture_cb to add the timestamp argument?

File: drivers/video/video.c [hunk 2]
```diff
@@ -1280,6 +1282,7 @@ static int video_qbuf(FAR struct video_mng_s *vmng,
             video_framebuff_get_vacant_container(&type_inf->bufinf);
           if (container != NULL)
             {
+              type_inf->seqnum = 0;
```
Comment (xiaoxiang781216) [inline drivers/video/video.c:1285]: ditto

File: drivers/video/video.c [hunk 3]
```diff
@@ -3260,6 +3262,7 @@ static int video_complete_capture(uint8_t err_code, uint32_t datasize)
     }
 
   type_inf->bufinf.vbuf_curr->buf.bytesused = datasize;
+  type_inf->bufinf.vbuf_curr->buf.timestamp = *ts;
```
[the remark this answers is not in the document]
Comment (PeterBee97) [reply drivers/video/video.c]: done

File: drivers/video/video.c [hunk 4]
```diff
@@ -701,6 +702,7 @@ static void change_video_state(FAR video_mng_t    *vmng,
               video_framebuff_get_vacant_container(&vmng->video_inf.bufinf);
       if (container != NULL)
         {
+          vmng->video_inf.seqnum = 0;
```
[the remark this answers is not in the document]
Comment (PeterBee97) [reply drivers/video/video.c]: done

File: drivers/video/video.c (hunk 2 above)
Comment (PeterBee97) [reply drivers/video/video.c]: done

File: drivers/video/video.c [hunk 5]
```diff
@@ -667,6 +669,7 @@ static int start_capture(enum v4l2_buf_type type,
   convert_to_imgsensorfmt(&fmt[VIDEO_FMT_SUB], &sf[IMGSENSOR_FMT_SUB]);
   convert_to_imgsensorinterval(interval, &si);
 
+  *seqnum = 0;
```
Comment (xiaoxiang781216) [inline drivers/video/video.c:672]: look like it's better to zero at caller site, sorry please ignore my previous comment.

Comment (PeterBee97) [reply drivers/video/video.c]: sure

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**